src/models/base.js

```diff
@@ -45,6 +45,11 @@ export class Model {
 	}
 
 	setProperties(properties) {
+		// Need to set validators before the rest of the properties
+		if (Object.keys(properties).find(key => key === 'validators')) {
+			this.set('validators', properties['validators']);
+		}
+
```

Comment:
This fixes the bug where a property could be set before its validator, if it has one.